src/System.Management.Automation/engine/NativeCommandParameterBinder.cs

```diff
@@ -218,13 +218,117 @@ private void appendOneNativeArgument(ExecutionContext context, object obj, char
+                                            }
+                                            else
+                                            {
+                                                if (! (pbo is System.IO.FileInfo || pbo is System.IO.DirectoryInfo))
+                                                {
+                                                    // If the object is not a filesystem object, then just append
+                                                    // the pattern unchanged
+                                                    _arguments.Append(arg);
+                                                    break;
+                                                }
+                                                first = false;
+                                            }
+                                            System.IO.FileInfo fi = pbo as System.IO.FileInfo;
+                                            if (fi != null)
+                                            {
+                                                _arguments.Append(computeRelativePath(cwd, fi.FullName));
+                                            }
+                                            else
+                                            {
+                                                System.IO.DirectoryInfo di = pbo as System.IO.DirectoryInfo;
+                                                if (di != null)
+                                                {
+                                                    _arguments.Append(computeRelativePath(cwd, di.FullName));
+                                                }
+                                            }
+                                        }
+                                    }
+                                    else
+                                    {
+                                        _arguments.Append(arg);
+                                    }
+                                }
+                                else
+                                {
+                                    _arguments.Append(arg);
+                                }
+                            }
+                            else
+                            {
+                                _arguments.Append(arg);
+                            }
+#else
                             _arguments.Append(arg);
+#endif
                         }
                     }
                 }
             } while (list != null);
         }
 
+#if UNIX
+        /// <summary>
+        /// Returns the path fragment truncated relative to the <paramref name="directory"/> parameter.
+        /// </summary>
+        /// <param name="directory">The directory to base the truncation on.</param>
+        /// <param name="path">The path to truncate.</param>
+        /// <returns>The relative path that was produced.</returns>
+        private static string computeRelativePath(string directory, string path)
```

Comment:
Why not use `SessionState.Path.NormalizeRelativePath.NormalizeRelativePath(string path, string basePath)`? We use it in `Resolve-Path`.